.github/workflows/ci.yml

```diff
@@ -41,8 +41,8 @@ jobs:
 
       - name: Install ⚙️
         run: |
-          python -m pip install --upgrade pip setuptools wheel
-          pip install --upgrade . -r dev-requirements.txt
+          python -m pip install --upgrade pip setuptools wheel pre-commit
+          # pip install --upgrade . -r requirements-dev.txt
```

Comment:
This should not be commented-out: the CI should check that this repo is an installable Python package.
Otherwise the `setup.py` might get broken by a PR without noone noticing.
Have you tried simply removing the dot? `pip install --upgrade -r requirements-dev.txt`

Reply:
The idea was to add the '-e .' to avoid repeating ressources from setup.py in requirements-dev.txt .

It seems that that ensures the installation of the package, because `pip install -r requirements-dev.txt` results in 

```
Installing collected packages: pre-commit-hooks
  Attempting uninstall: pre-commit-hooks
    Found existing installation: pre-commit-hooks 1.1.13
    Uninstalling pre-commit-hooks-1.1.13:
      Successfully uninstalled pre-commit-hooks-1.1.13
  Running setup.py develop for pre-commit-hooks
Successfully installed pre-commit-hooks-1.1.13
```

after a `pip install --upgrade .`

So I updated the ci to do both steps and the upgrade last.